marklogic/cloud_auth.py

```diff
@@ -0,0 +1,25 @@
+from urllib.parse import urljoin
+
+import requests
+from requests.auth import AuthBase
+
+# See https://requests.readthedocs.io/en/latest/user/advanced/#custom-authentication
+
+
+class MarkLogicCloudAuth(AuthBase):
+    def __init__(self, base_url: str, api_key: str, verify):
+        self._base_url = base_url
+        self._verify = verify
+        self._generate_token(api_key)
+
+    def _generate_token(self, api_key: str):
+        response = requests.post(
+            urljoin(self._base_url, "/token"),
+            data={"grant_type": "apikey", "key": api_key},
+            verify=self._verify,
+        )
+        self._access_token = response.json()["access_token"]
```

Comment:
Do you need any error handling here?

Reply:
Good point, I forget if a 4xx gets thrown right away - I'll add a test case for it.